virttest/env_process.py

```diff
@@ -98,10 +99,39 @@ def preprocess_vm(test, params, env, name):
     :param env: The environment (a dict-like object).
     :param name: The name of the VM object.
     """
+    def validate_memory_resource():
+        """Validate host has enough memory to lauch VMs"""
+        vms = set(params.objects("vms")) - set(env.get_all_vms())
+        magnification, requried_mem, count = 1.0, 0, 1
+        for vm_name in vms:
+            vm_params = params.object_params(vm_name)
+            if vm_params.get("start_vm") == "yes":
+                mem = "%sM" % vm_params.get("mem", 512)
+                requried_mem += float(utils_misc.normalize_data_size(
+                    mem, order_magnitude="K"))
+                count += 1
+        if (params.get("setup_ksm") == "yes" and
+                params.get("ksm_run", "1") == "1"):
+            magnification = 1.2
+        provide_mem = "%s KB" % (memory.freememtotal() * magnification)
+        provide_mem = float(
+            utils_misc.normalize_data_size(provide_mem))
```

Comment:
just want to specify, required_mem and provide_mem use different units to compare.